### ISSUE TITLE
Another question on CERT

### ISSUE THREAD

**Issue by sachabarber** [wsilveiranz/EventHubsKafkaSample#2]
So I have read your excellent blog post (https://notetoself.tech/2018/06/03/acessing-event-hubs-with-confluent-kafka-library/) on this, and also seen the other issue around Cert, and saw you got the PEM file from : https://curl.haxx.se/docs/caextract.html

But am I being a bit thick here, but that PEM file contains about 138 individual certificates, which one WILL it / DOES it use for the actual Kafka/EventHub SSL comms?

Should there not be a specific one that it uses, like a single PEM file that was created specifically for this Kafka/EventHub comms?

**Comment by wsilveiranz** [comment]
Hi Sacha,

The PEM file contains a list of  Certificate Authorities (CA) root certificates. Since we don't know which certificate Event Grid will be using - they rotate certs and don't keep within a single CA - it is better to get a list that has already been provided than trying to create one. And since we are only using that list to validate that the certificate is coming from a valid CA, it is ok to use that PEM file. If you want you can try to bring it down to the list of CA certificates that Azure uses.

So in summary this PEM file doesn't hold the SSL certificate that the EventHub uses, but just a list of CA root certificates.

**Comment by sachabarber** [comment]
Ok thanks for the answer I will close this one
